### PR TITLE
Implement send with new protocol buffers

### PR DESCRIPTION
**Pull request by keefertaylor**
## High Level Overview of Change

Implement `send` for the  `XpringClient` which uses official rippled protocol buffers. 

Add unit and integration tests. The unit tests are taken from the `LegacyDefaultXpringClient` tests, which ensure that the implementation behaves consistently across protocol buffers. 

Analog in JS: https://github.com/xpring-eng/Xpring-JS/pull/136

### Context of Change

rippled implemented protocol buffer support natively, which uses a new and incompatible set of protocol buffers. The implementation in rippled was completed in: https://github.com/ripple/rippled/pull/3159.

Protocol buffer upgrades are being done in piecemeal fashion for ease of review. 

*Status*

- [x] [Generate new versions of the protocol buffers](https://github.com/xpring-eng/XpringKit/pull/68)
- [x] [Migrate existing code to a `legacy` subdirectory and namespace](https://github.com/xpring-eng/XpringKit/pull/72)
- [x] Create code that works with the new protocol buffers from the rippled team
   - [x] [Create a new `NetworkClient` with fakes](https://github.com/xpring-eng/XpringKit/pull/86)
   - [x] [Create a new `DefaultXpringClient` with tests](https://github.com/xpring-eng/XpringKit/pull/87)
   - [x] Read only calls for new `DefaultXpringClient` (many PRs)
   - [x] Write calls for `DefaultXpringClient` (this PR)
- [x] [Add a boolean parameter to switch between implementations](https://github.com/xpring-eng/XpringKit/pull/90)

### Type of Change

- [x] New feature (non-breaking change which adds functionality)

## Before / After

Users who use the new protocol buffers will be able to send XRP.
## Test Plan

Unit tests on CI continue to pass.  Integration test passes.

### CODE REVIEW

**Comment by keefertaylor** [comment]
Pulling this PR back as this PR is based off #98

File: Tests/DefaultXpringClientTest.swift [hunk 1]
```diff
@@ -68,6 +68,124 @@ final class DefaultXpringClientTest: XCTestCase {
...
+  func testSendWithClassicAddress() {
+    // GIVEN a classic address.
+    guard let classicAddressComponents = Utils.decode(xAddress: .destinationAddress) else {
+      XCTFail("Failed to decode X - Address.")
+      return
+    }
+    let xpringClient = DefaultXpringClient(networkClient: FakeNetworkClient.successfulFakeNetworkClient)
+
+    // WHEN XRP is sent to a classic address THEN an error is thrown.
+    XCTAssertThrowsError(try xpringClient.send(
```
Comment (austin-king) [inline Tests/DefaultXpringClientTest.swift]: Why are we cutting functionality to send to classic addresses? Just curious

File: Tests/DefaultXpringClientTest.swift [hunk 2]
```diff
@@ -68,6 +68,124 @@ final class DefaultXpringClientTest: XCTestCase {
...
+  }
+
+  func testSendWithLatestLedgerSequenceFailure() {
+    // GIVEN a Xpring client which will fail to return the latest validated ledger sequence.
+    let networkClient = FakeNetworkClient(
+      accountInfoResult: .success(.testGetAccountInfoResponse),
+      feeResult: .failure(XpringKitTestError.mockFailure),
+      submitTransactionResult: .success(.testSubmitTransactionResponse),
+      transactionStatusResult: .success(.testGetTxResponse)
+    )
```
Comment (austin-king) [inline Tests/DefaultXpringClientTest.swift]: Isn't this the same FakeNetworkClient config as the above test `testSendWithFeeFailure`? Looks like both have the feeResult field set to .failure with the rest as success.

File: Tests/DefaultXpringClientTest.swift (hunk 1 above)
Comment (keefertaylor) [reply Tests/DefaultXpringClientTest.swift]: This was a product level decision that we wanted people to use the newest features the ecosystem had to offer.

File: Tests/DefaultXpringClientTest.swift (hunk 2 above)
Comment (keefertaylor) [reply Tests/DefaultXpringClientTest.swift]: Ah, my fault. 

In the old protocol buffers, the fee / last ledger were different RPCs but now they're the same.

**Pull request merged**